### PR TITLE
Energy Weapons

### PR DESCRIPTION
**Pull request by Lamasmaster**
## About The Pull Request
<details>
<summary>
	About The Pull Request
</summary>
<hr>

Removes Serenity (reduced to oddity spawn), Welder Laser, Mindflayer, 3 tesla shotguns besides the base one, Eclipse/Aster, Star/Blue Star
Renames several energy weapons and most of mech energy weapons (MHM - Minerva Heavy Manufacturing / MHM-MS - Minerva Heavy Manufacturing - Milispec / PHM - Phobos Hellion Metalworks)
Replaces sounds on a lot of the energy weapons to add to them not being same exact weapon copy and pasted (even if this is sort of a case) ((taken from F13))
X-ray = Now pierces walls
EBR (Nuclear Advanced Energy Gun formerly) = Has a tazer mode and beam mode which irradiates it's targets 
Heavy Pulse Rifle = deals more damage alongside it's mech counterpart but unlike it produces 1 1 EMP
Mech Ion Cannon = Increased radius to make it actually better than the handheld version like the comment would imply
Pulse Cannon = Now has burst mode like scorcher (stacked lasers)
Firestorm SMG = Now ignites it's targets
Plasma Weapons (across the board) = Shots now ignite the target since getting hit by a superheated projectile might just do that
Moved energy sounds from general folder to it's own folder under weapons to make it bit easier to look at what is what
Fixed most of the missing onmob sprites

<hr>
</details>

## Changelog
:cl:
del: Obsolete Energy Weapons
balance: Energy weapons
fix: Energy weapon sprites
soundadd: Energy weapon sounds
refactor: Energy weapon sounds
refactor: Energy weapon sorting
/:cl:

### CODE REVIEW

**Comment by Trilbyspaceclone** [comment]
Laser Core, Antebellum should stay rest I an aggree/ understand

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 1]
```diff
@@ -1,11 +1,13 @@
 /obj/item/gun/energy/gun/nuclear
-	name = "Prototype: advanced energy gun"
-	desc = "An energy gun with an experimental miniaturized reactor with a reserved spot for a scope to be attached. It offers a self charging cell and fires in both lethal and stun."
+	name = "\"EBR-IV\" Nuclear Particle Rifle"
+	desc = "A stolen prototype of a 'self-sustaining' experimental breeder rifle feeding itself off a built-in nuclear reactor. Projecting either high powered tazer shots or deadly particle beams at will of it's user. \
```
Comment (Trilbyspaceclone) [inline code/modules/projectiles/guns/energy/nuclear.dm:3]:
```suggestion
	desc = "A prototype of a 'self-sustaining' experimental breeder rifle feeding itself off a built-in nuclear reactor. Projecting either high powered tazer shots or deadly particle beams at will of it's user. \
```

Comment (Lamasmaster) [reply code/modules/projectiles/guns/energy/nuclear.dm]: Changed

File: code/modules/projectiles/projectile/beams.dm [hunk 2]
```diff
@@ -81,6 +81,7 @@ In pvp they also have more lasting damages, such as infections, and pain form bu
 /obj/item/projectile/beam/weak/smg
 	damage_types = list(BURN = 10)
 	added_damage_laser_pve = 10
+	fire_stacks = 2 //So it doesn't become dumb fast
```
[the remark this answers is not in the document]
Comment (Lamasmaster) [reply code/modules/projectiles/projectile/beams.dm:88]: Made it so firestorm has it's own beam that does this I forgor about mobs.

**Pull request merged**